src/components/pages/search-algolia/Filters/Filters.jsx

```diff
@@ -0,0 +1,202 @@
+import { Criteria } from '../Criteria/Criteria'
+import { Route, Switch } from 'react-router'
+import { GEOLOCATION_CRITERIA } from '../Criteria/criteriaEnums'
+import { checkIfAroundMe } from '../utils/checkIfAroundMe'
+
+export class Filters extends PureComponent {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentFilters: {
+        categories: props.initialFilters.categories,
+        isSearchAroundMe: props.initialFilters.isSearchAroundMe,
+        sortCriteria: props.initialFilters.sortCriteria
+      },
+      initialFilters: {
+        categories: props.initialFilters.categories,
+        isSearchAroundMe: props.initialFilters.isSearchAroundMe,
+        sortCriteria: props.initialFilters.sortCriteria
+      },
+      offers: props.offers
+    }
+  }
+
+  fetchOffers = (keywords, geolocation = null, categories, indexSuffix) => {
+    const { showFailModal } = this.props
+    fetchAlgolia(keywords, 0, geolocation, categories, indexSuffix)
+      .then(offers => {
+        this.setState({
+          offers: offers
+        })
+      })
+      .catch(() => {
+        showFailModal()
+      })
+  }
+
+  handleGeolocationCriterionSelection = criterionKey => () => {
+    const {
+      isUserAllowedToSelectCriterion,
+      isGeolocationEnabled,
+      redirectToSearchFiltersPage
+    } = this.props
+    const { currentFilters } = this.state
+
+    if (!isUserAllowedToSelectCriterion(criterionKey, isGeolocationEnabled)) return
+
+    this.setState({
+      currentFilters: {
+        ...currentFilters,
+        isSearchAroundMe: GEOLOCATION_CRITERIA[criterionKey].label === GEOLOCATION_CRITERIA.AROUND_ME.label
+      }
+    }, () => {
+      this.process()
+      redirectToSearchFiltersPage()
+    })
+  }
+
+  resetFilters = () => {
+    const { initialFilters } = this.state
+
+    this.setState({
+      currentFilters: {
+        ...initialFilters
+      },
+    }, () => {
+      this.process()
+    })
+  }
+
+  process = () => {
+    const { history, geolocation, query } = this.props
+    const { currentFilters } = this.state
+    const queryParams = query.parse()
+    const keywords = queryParams['mots-cles'] || ''
+
+    const { categories, isSearchAroundMe, sortCriteria} = currentFilters
+    isSearchAroundMe ?
+      this.fetchOffers(keywords, geolocation, categories, sortCriteria) :
+      this.fetchOffers(keywords, null, categories, sortCriteria)
+
+    const autourDeMoi = checkIfAroundMe(currentFilters.isSearchAroundMe)
+    const category = currentFilters.categories !== '' ? currentFilters.categories.join(';') : ''
+    const tri = currentFilters.sortCriteria
+    history.replace({
+      search: `?mots-cles=${keywords}&autour-de-moi=${autourDeMoi}&tri=${tri}&categories=${category}`
+    })
+  }
+
+  handleFilterOffers = () => {
+    const { history, getFilteredOffers } = this.props
+    const { location: { search = '' } } = history
+    const { offers } = this.state
+    getFilteredOffers(offers)
```

Comment:
J'ai l'impression que c'est une fonction qui a des effets de bord, avec un `this.setState` qui vient d'un autre composant (`SearchResults` ?). Ça crée un couplage fort entre les deux composants, c'est voulu ?

Reply:
En tous cas, je renommerais bien pour faire en sorte qu'on voit qu'il y a un effet de bord
```suggestion
    updateFilteredOffers(offers)
```

Reply:
C'est fait exprès. Le composant Filters renvoie les résultats filtrés au parent SearchResults

Reply:
👍